### DISCUSSION TITLE
Allow the github-tags datasource to be configured for different credentials

### DISCUSSION THREAD

**Discussion by KnisterPeter** [renovatebot/renovate#30598] [Suggest an Idea]
### Tell us more.

Currently it's possible to configure renvoate to use different credentials for different repositories.
See https://docs.renovatebot.com/getting-started/private-packages/#github-and-enterprise-repo-scoped-credentials

This is not possible for the `github-tags` datasource, as it uses GraphQL and the API URL is identical for all requests.
It would be nice to send different credentials per org and/or repo.

**Comment by viceice** [comment]
it's possible, see

https://github.com/renovatebot/renovate/blob/4c99aea3ef33025a7fc12abd847d6e34e5f03322/lib/util/http/github.ts#L289

**Reply by KnisterPeter** [reply]
Yes, but that's not for the graphql part. The GraphQL Endpoint is always https://api.github.com/graphql.

**Reply by viceice** [reply]
it works, you need to configure a host rule like in the linked snippet

**Reply by KnisterPeter** [reply]
Them I'm not sure why we see this warning in our logs:

```
 WARN: Package lookup failures (repository=org/repo, branch=renovate/remix-i18next-6.x)
       "warnings": ["Failed to look up github-tags package org/repo2"],
       "files": [".github/workflows/auto-pr.yml", ".github/workflows/release.yml"]

```

Our host rule looks like this and configures the token for all requests to the required org.

```
  hostRules: [
    {
      hostType: "github",
      matchHost: "https://api.github.com/repos/org",
      token: process.env.TOKEN,
    },
  ],
```

Does it only work for a specific combination of Owner and Repo? I wanted to configure all requests to a specific Owner with it's own token. For the HTTP API calls it works fine like that.

**Reply by viceice** [reply]
can you try to remove hostType? did you validated the token is correctly passed?

**Reply by viceice** [reply]
are you self hosting? I think `TOKEN` is a reserved environment variable and will probably override other host rules
